sumy/summarizers/text_rank.py

```diff
@@ -64,9 +64,11 @@ def _create_matrix(self, document):
         sentences_count = len(sentences_as_words)
         weights = numpy.zeros((sentences_count, sentences_count))
 
-        for i, words_i in enumerate(sentences_as_words):
-            for j, words_j in enumerate(sentences_as_words):
-                weights[i, j] = self._rate_sentences_edge(words_i, words_j)
+        for i in range(0, sentences_count-1):
+            for j in range(i+1, sentences_count):
```

Comment:
Are you sure you don't want `range(i, sentences_count)`? Aren't the words at positions `[i, i]` missing then?

Reply:
They are but they are not relevant (similarity to yourself stays 0 which is OK) and does not change page rank.